indexer/editable_map_object.cpp

```diff
@@ -743,38 +742,30 @@ bool EditableMapObject::ValidateName(string const & name)
   if (name.empty())
     return true;
 
-  if (strings::IsASCIIString(name))
-  {
-    static auto const s_nameRegex = regex(R"(^[ A-Za-z0-9.,?!@°#$%&()\-\+\/\\\[\]:;"'`]+$)");
-    return regex_match(name, s_nameRegex);
-  }
+  static std::u32string_view constexpr excludedSymbols = U"^§><*=_±√•÷×¶";
 
-  std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> converter;
-
-  std::u32string u32name;
-  try
-  {
-    u32name = converter.from_bytes(name);
-  }
-  catch (std::range_error const &)
-  {
-    // Cannot convert, for ex. it is possible for some emoji.
-    return false;
-  }
-
-  std::u32string const excludedSymbols = U"^~§><{}*=_±\n\t\r\v\f|√•÷×¶";
-
-  for (auto const ch : u32name)
+  using Iter = utf8::unchecked::iterator<string::const_iterator>;
+  for (Iter it{name.cbegin()}; it != Iter{name.cend()}; ++it)
   {
+    auto const ch = *it;
+    // Exclude ASCII control characters.
+    if (ch < U'\U0000001F')
```

Comment:
nit: Why so complicated constants instead of simple 0x1F, 0x1F9FF, .. here and below?

Reply:
`U'\U1F'` -> error: incomplete universal character name

Reply:
```suggestion
    if (ch <= U'\U0000001F')
```

Reply:
? Just ```if(ch <= 0x1F)```